### PR TITLE
v1.0.0 deprecations

### PR DESCRIPTION
**Pull request by edenhill**
See #2020

### CODE REVIEW

File: CONFIGURATION.md [hunk 1]
```diff
@@ -95,11 +95,11 @@ max.partition.fetch.bytes                |  C  |                 |
...
 consume_cb                               |  C  |                 |               | low        | Message consume callback (set with rd_kafka_conf_set_consume_cb()) <br>*Type: pointer*
 rebalance_cb                             |  C  |                 |               | low        | Called after consumer group has been rebalanced (set with rd_kafka_conf_set_rebalance_cb()) <br>*Type: pointer*
 offset_commit_cb                         |  C  |                 |               | low        | Offset commit result propagation callback. (set with rd_kafka_conf_set_offset_commit_cb()) <br>*Type: pointer*
-enable.partition.eof                     |  C  | true, false     |          true | low        | Emit RD_KAFKA_RESP_ERR__PARTITION_EOF event whenever the consumer reaches the end of a partition. <br>*Type: boolean*
+enable.partition.eof                     |  C  | true, false     |         false | low        | Emit RD_KAFKA_RESP_ERR__PARTITION_EOF event whenever the consumer reaches the end of a partition. <br>*Type: boolean*
```
Comment (mhowlett) [inline CONFIGURATION.md]: 🎉

File: CONFIGURATION.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ internal.termination.signal              |  *  | 0 .. 128        |             0
...
+broker.version.fallback                  |  *  |                 |        0.10.0 | medium     | Older broker versions (before 0.10.0) provide no way for a client to query for supported protocol features (ApiVersionRequest, see `api.version.request`) making it impossible for the client to know what features it may use. As a workaround a user may set this property to the expected broker version and the client will automatically adjust its feature set accordingly if the ApiVersionRequest fails (or is disabled). The fallback broker version will be used for `api.version.fallback.ms`. Valid values are: 0.9.0, 0.8.2, 0.8.1, 0.8.0. Any other value, such as 0.10.2.1, enables ApiVersionRequests. <br>*Type: string*
```
Comment (rnpridgeon) [inline CONFIGURATION.md]: 🥇

File: src/rdkafka.h [hunk 3]
```diff
@@ -3753,6 +3761,8 @@ void rd_kafka_set_logger(rd_kafka_t *rk,
  *
  * If the \p \"debug\" configuration property is set the level is automatically
  * adjusted to \c LOG_DEBUG (7).
+ *
+ * @deprecated Set the \c "log_level" configuration property instead.
```
Comment (rnpridgeon) [inline src/rdkafka.h]: Should the debug configuration in Configuration.md as well?

File: tests/0061-consumer_lag.cpp [hunk 4]
```diff
@@ -131,8 +131,7 @@ static void do_test_consumer_lag (void) {
   if (conf->set("event_cb", &stats, errstr) != RdKafka::Conf::CONF_OK)
     Test::Fail("set event_cb failed: " + errstr);
   Test::conf_set(conf, "group.id", topic);
-  Test::conf_set(conf, "enable.auto.commit", "false");
-  Test::conf_set(conf, "enable.partition.eof", "false");
+  Test::conf_set(conf, "enable.auto.commit", "false"); /* now default */
```
Comment (rnpridgeon) [inline tests/0061-consumer_lag.cpp]: Configuraiton.md shows the default for `enable.auto.commit` as true

File: src/rdkafka.h (hunk 3 above)
Comment (edenhill) [reply src/rdkafka.h]: Should debug as well what? Be deprecated? No, it is very much needed still (still a few more bugs in the code somewhere!)

File: tests/0061-consumer_lag.cpp (hunk 4 above)
Comment (edenhill) [reply tests/0061-consumer_lag.cpp]: Whoops, good spotting!

File: src/rdkafka.h (hunk 3 above)
Comment (rnpridgeon) [reply src/rdkafka.h]: I misinterpreted what was being said. Carry on

**Pull request merged**